### ISSUE TITLE
Typos in IMAP search methods

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4028]
|<img alt="Magneto's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Magneto reported|
|-|-|
|Trac ID|trac#4028|
|Type|defect|
|Created|2009-09-15 06:51:40Z|

def search_SENTBEFORE(self, query, id, msg):
      date = msg.getHeader(False, 'date').get('date', '')

def search_SENTON(self, query, id, msg):
      date = msg.getHeader(False, 'date').get('date', '')

def search_SENTSINCE(self, query, id, msg):
      date = msg.getHeader(False, 'date').get('date', '')

It should be msg.getHeaders, (missing "s"). duh....
    

Attachments:

* [svn_fix_search_typos.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bff/bff768c7aea91be241e873f1e10704cf27740f19/16ca302e0f0173050f16e70b4712ddb750aaf9d1.txt) (875 bytes) - added by Magneto on 2009-10-09 19:41:03Z - svn_diff_fix_typos_search
* [getHeaders.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bff/bff768c7aea91be241e873f1e10704cf27740f19/a295f662e3bf691784461c91ea29c47a000e30a9.patch) (4524 bytes) - added by jesstess on 2009-12-16 21:41:10Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4028 4028
type__defect defect
reporter__Magneto Magneto
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__imap imap
time__1252997500000000 1252997500000000
changetime__1261003541000000 1261003541000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess removed owner|
|-|-|

In getHeaders.patch: changed getHeader to getHeaders in the 3 cases in the ticket description and added a test class that exercises those 3 functions.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @jesstess|
|-|-|

A few simple points:

  1. In light of [#4174](https://github.com/twisted/twisted/issues/4174), maybe renaming or removing the `id` attribute would be prudent.
  1. Some extra spaces are needed between test methods and the test case and the following class (two between methods, three between classes)
  1. Docstrings for the three modified `search_*` methods would be great - particularly discussing the value expected to be at the beginning of `query`.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess removed owner|
|-|-|

Points 1-3 addressed in the updated patch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @jesstess|
|-|-|

  1. The last new search test method is still using `self.id` (and passing anyway, since the search methods don't care about the id at all).
  1. For the search_* docstrings, `msg` actually needs to be a _provider_ of `IMessage` - generally classes implement, instances provide.

Otherwise looks good.  Please apply when those two things are fixed.  Thanks. :)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess set status to `closed`|
|-|-|

(In [27797]) Apply getHeaders.patch

Author: jesstess
Reviewer: exarkun
Fixes: [#4028](https://github.com/twisted/twisted/issues/4028)

Change calls in 3 search_* methods of IMAP4Server from the non-existent
'getHeader()' to 'getHeaders()', and add docstrings clarifying the behavior of
these search methods.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
